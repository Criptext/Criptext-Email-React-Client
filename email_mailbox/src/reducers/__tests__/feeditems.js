/* eslint-env node, jest */

import feedItemsReducer from './../feeditems';
import {
  addFeedItems,
  updateFeedItemSuccess,
  removeFeedItemSuccess
} from './../../actions/index';
import json from './../../../public/feeds.json';

jest.mock('./../../utils/ipc');
jest.mock('./../../utils/electronInterface');
jest.mock('./../../utils/electronEventInterface');

describe('Feed actions: ', () => {
  const { feeds } = json;
  const initState = feeds => feedItemsReducer(undefined, addFeedItems(feeds));

  it('Add FeedItems to state', () => {
    expect(initState(feeds)).toMatchSnapshot();
  });

  it('Add feedItems to state. clear: true', () => {
    const prevState = initState(feeds);
    const newFeedItems = {
      0: {
        id: 0,
        date: '2018-08-17T22:11:29.754Z',
        type: '1',
        location: undefined,
        seen: false,
        emailId: 1,
        contactId: 2,
        fileId: undefined,
        isNew: true
      }
    };
    const clear = true;
    const action = addFeedItems(newFeedItems, clear);
    const nextState = feedItemsReducer(prevState, action);
    expect(nextState).toMatchSnapshot();
  });

  it('Update field by FeedItem id', () => {
    const idToUpdate = 1;
    const prevState = initState(feeds);
    const action = updateFeedItemSuccess({
      feedItemId: idToUpdate,
      seen: true
    });
    const nextState = feedItemsReducer(prevState, action);
    expect(nextState.get(`${idToUpdate}`).get('seen')).toBe(true);
  });

  it('Remove FeedItem by id', () => {
    const idToDelete = 3;
    const prevState = initState(feeds);
    const action = removeFeedItemSuccess(idToDelete);
    const nextState = feedItemsReducer(prevState, action);
    expect(nextState.get(`${idToDelete}`)).toBe(undefined);
  });
});
