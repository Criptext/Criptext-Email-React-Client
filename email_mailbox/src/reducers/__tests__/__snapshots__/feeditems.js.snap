// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed actions:  Add FeedItems to state 1`] = `
Immutable.Map {
  "badge": 0,
  "list": Immutable.Map {
    "1": Immutable.Map {
      "id": "1",
      "date": "2018-06-23T22:11:29.754Z",
      "type": "1",
      "location": "undefined",
      "seen": false,
      "emailId": 1,
      "contactId": 2,
      "fileId": "undefined",
    },
    "2": Immutable.Map {
      "id": "2",
      "date": "2018-06-23T22:11:29.754Z",
      "type": "1",
      "location": "undefined",
      "seen": false,
      "emailId": 2,
      "contactId": 2,
      "fileId": "undefined",
    },
    "3": Immutable.Map {
      "id": "3",
      "date": "2018-06-23T22:11:29.754Z",
      "type": "1",
      "location": "undefined",
      "seen": false,
      "emailId": 3,
      "contactId": 2,
      "fileId": "undefined",
    },
  },
}
`;

exports[`Feed actions:  Add feedItems to state. clear: true 1`] = `
Immutable.Map {
  "badge": 0,
  "list": Immutable.Map {
    "0": Immutable.Map {
      "id": 0,
      "date": "2018-08-17T22:11:29.754Z",
      "type": "1",
      "location": undefined,
      "seen": false,
      "emailId": 1,
      "contactId": 2,
      "fileId": undefined,
    },
  },
}
`;
